src/controllers/connection/newConnection.ts

```diff
@@ -43,31 +48,28 @@ export class NewConnectionController extends HTMLController {
    */
   @SuccessResponse(200)
   @Get('/verify-company')
-  public async verifyCompanyForm(@Query() companyNumber: string): Promise<HTML> {
-    this.logger.debug('connections page requested')
-    const regex =
-      /^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\d{2})\d{6})|((IP|SP|RS)[A-Z\d]{6})|(SL\d{5}[\dA]))$/
-    if (!regex.test(`${companyNumber}`)) {
-      return this.html(
-        this.newConnection.companyNumberInput({
-          targetBox: await this.newConnection.companyEmptyTextBox({ errorMessage: 'Company number format incorrect' }),
-        })
-      )
-    }
-    let company
+  public async verifyCompanyForm(@Query() companyNumber: COMPANY_NUMBER, company?: CompanyProfile): Promise<HTML> {
```

Comment:
Adding the company here with a zod type generates this error https://github.com/digicatapult/veritable-ui/actions/runs/9286851097/job/25554482149#:~:text=There%20was%20a,in%20%27NewConnectionController.verifyCompanyForm%27